ddtrace/propagation/http.py

```diff
@@ -815,8 +821,13 @@ def _inject(span_context, headers):
             headers[_HTTP_HEADER_TRACEPARENT] = tp
             # only inject tracestate if traceparent injected: https://www.w3.org/TR/trace-context/#tracestate-header
             ts = span_context._tracestate
-            if ts:
-                headers[_HTTP_HEADER_TRACESTATE] = ts
+            # Adds last datadog parent_id to tracestate. This tag is used to reconnect a traces with missing spans
+            if "dd=" in ts:
+                ts = ts.replace("dd=", "dd=p:{:016x};".format(span_context.span_id or 0))
+            else:
+                ts = "dd=p:{:016x}".format(span_context.span_id or 0)
```

Comment:
Is this just accounting for cases where there are multiple occurrences of `dd=` in the if case, and empty/malformed tracestates in the else case?

Reply:
Having the else clause allows us to account for the case where `dd=` is not in the tracestate but this case will only occur if there is a bug in our tracer (spans do not have sampling decsions or tracestate is malformed).